qiskit/algorithms/minimum_eigen_solvers/adapt_vqe.py

```diff
@@ -0,0 +1,342 @@
+        match = cycle_regex.search(" ".join(map(str, indices)))
+        logger.debug("Cycle detected: %s", match)
+        logger.info("Alternating sequence found. Finishing.")
+        # Additionally we also need to check whether the last two numbers are identical, because the
+        # reg-ex above will only find cycles of at least two consecutive numbers.
+        # It is sufficient to assert that the last two numbers are different due to the iterative
+        # nature of the algorithm.
+        return match is not None or (len(indices) > 1 and indices[-2] == indices[-1])
+
+    @staticmethod
+    def _log_gradients(
+        iteration: int,
+        cur_grads: Tuple[float, PauliSumOp],
+        max_grad: Tuple[float, PauliSumOp | None],
+    ):
+        if logger.isEnabledFor(logging.INFO):
+            gradlog = f"\nGradients in iteration #{str(iteration)} "
+            gradlog += "\nID: Excitation Operator: Gradient  <(*) maximum>"
+            for i, grad in enumerate(cur_grads):
+                gradlog += f"\n{str(i)}: {str(grad[1])}: {str(grad[0])}"
+                if grad[1] == max_grad[1]:
+                    gradlog += "\t(*)"
+            logger.info(gradlog)
+
+    def compute_minimum_eigenvalue(
+        self,
+        operator: OperatorBase,
+        aux_operators: Optional[ListOrDict[PauliSumOp]] = None,
+    ) -> AdaptVQEResult:
+        """Computes the ground state.
+
+        Args:
+            operator: Operator to evaluate
+            aux_operators: Additional auxiliary operators to evaluate.
+
+        Raises:
+            QiskitError: If a solver other than VQE
+                (`qiskit.algorithms.minimum_eigen_solvers.vqe.py`) or a ansatz other than
+                EvolvedOperatorAnsatz (`qiskit.circuit.library.evolved_operator_ansatz.py`)
+                is provided or if the algorithm finishes due to an unforeseen reason.
+            ValueError: If the grouped property object returned by the driver does not contain a
+                main property as requested by the problem being solved (`problem.main_property_name`)
+            QiskitError: If the user-provided `aux_operators` contain a name which clashes
+                with an internally constructed auxiliary operator. Note: the names used for the
+                internal auxiliary operators correspond to the `Property.name` attributes which
+                generated the respective operators.
+            QiskitError: If the chosen gradient method appears to result in all-zero gradients.
+
+        Returns:
+            An AdaptVQEResult (`qiskit.algorithms.minimum_eigen_solvers.adapt_vqe.AdaptVQEResult`)
+            which is a VQEResult (`qiskit.algorithms.minimum_eigen_solvers.vqe.VQEResult`) but also
+            includes runtime information about the AdaptVQE algorithm like the number of iterations,
+            finishing criterion, and the final maximum gradient.
+        """
+        if not isinstance(self._tmp_ansatz, EvolvedOperatorAnsatz):
```

Comment:
If we follow my suggestion above, and remove the `ansatz` argument, we should here access `self.solver.ansatz` instead. This would allow "lazier" construction of this algorithm